src/type-checking.md

```diff
@@ -1,10 +1,10 @@
 # Type checking
 
-The [`rustc_typeck`][typeck] crate contains the source for "type collection"
+The [`hir_analysis`] crate contains the source for "type collection"
 and "type checking", as well as a few other bits of related functionality. (It
 draws heavily on the [type inference] and [trait solving].)
```

Comment:
this isn't completely right as we've moved type check to `rustc_hir_typeck`.

maybe something like
```suggestion
The [`hir_analysis`] crate contains the source for "type collection" as well as a bunch of related functionality. Checking the bodies of functions is implemented in the [`hir_typeck`] crate. (It draws heavily on the [type inference] and [trait solving].)
```
```

Reply:
which of these 2 crates "draws heavily..."

Reply:
both do :grin: